ion-schema/src/isl/ranges.rs

```diff
@@ -0,0 +1,471 @@
+use crate::isl::util::TimestampPrecision;
+use crate::{invalid_schema_error, invalid_schema_error_raw, isl_require};
+use crate::{IonSchemaResult, WriteToIsl};
+use ion_rs::element::writer::ElementWriter;
+use ion_rs::element::Element;
+use ion_rs::Decimal;
+use ion_rs::{IonType, IonWriter, Timestamp};
+use std::fmt::{Display, Formatter};
+
+/// The end of a [`Range`].
+#[derive(Debug, PartialEq, Clone)]
+pub enum Limit<T> {
+    Unbounded,
```

Comment:
What does unbound represent here? It would be nice to add some document comments for these enum variants and how they are used.

Reply:
I still think we should add some comments about what `Unbounded` represents here and when to use it? I see that you have sued it to represent `min`/`max` but it would be helpful to add some doc comments for it so the usecase is clear.